extensions/GitExtension/FileExplorerGitIntegration/Models/GitRepositoryStatus.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using LibGit2Sharp;
+
+namespace FileExplorerGitIntegration.Models;
+
+internal sealed class GitRepositoryStatus
+{
+    private readonly Dictionary<string, GitStatusEntry> _entries = new();
+    private readonly List<GitStatusEntry> _added = new();
+    private readonly List<GitStatusEntry> _staged = new();
+    private readonly List<GitStatusEntry> _removed = new();
+    private readonly List<GitStatusEntry> _untracked = new();
+    private readonly List<GitStatusEntry> _modified = new();
+    private readonly List<GitStatusEntry> _missing = new();
+    private readonly List<GitStatusEntry> _ignored = new();
+    private readonly List<GitStatusEntry> _renamedInIndex = new();
+    private readonly List<GitStatusEntry> _renamedInWorkDir = new();
+    private readonly List<GitStatusEntry> _conflicted = new();
+
+    public GitRepositoryStatus()
+    {
+    }
+
+    public void Add(string path, GitStatusEntry status)

```

Comment:
If you want to be fancy you could compress this into a loop.
1. Instead of individual lists from line 11 to 20, make a Dictionary<StatusFlag, List<GitStatusEntry>> I don't know what the type of status.Status.Flag is.
2. Run through all keys with

    if(status.Status.HasFlag(key)
    {
        value.Add(status);
    }

Reply:
> ... I don't know what the type of status.Status.Flag is.

`status.Status` is a ["flags" enum](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/enum#enumeration-types-as-bit-flags).